lib/zinnia.js

```diff
@@ -76,6 +76,11 @@ const updateAllSourceFiles = async ({
             onFailedAttempt: err => {
               console.error(err)
               console.error(`Failed to download ${module} source. Retrying...`)
+              if (String(err).includes('You are being rate limited')) {
+                // Don't DDOS the w3name services
+                console.error('Rate limited. Waiting 60 seconds...')
+                return timers.setTimeout(60_000)
```

Comment:
I propose to randomise the timeout, e.g. pick a random number between `30_000` and `90_000`.

Use case:
1. I run 100 Station nodes and (re)start them roughly at the same time
2. 10 nodes resolve w3name, 90 nodes hit the rate limit
3. 60 seconds later, 90 nodes retry the w3name request
4. 10 nodes succeed, 80 nodes hit the rate limit again
5. (...)

The randomised delay should better spread the load of w3name requests into a wider time interval.

Reply:
good idea!